sw/device/silicon_creator/lib/dice.c

```diff
@@ -319,3 +353,77 @@ rom_error_t dice_cdi_1_cert_build(manuf_certgen_inputs_t *inputs,
 
   return kErrorOk;
 }
+
+rom_error_t dice_tpm_ek_cert_build(manuf_certgen_inputs_t *inputs,
+                                   attestation_public_key_t *tpm_ek_pubkey,
+                                   uint8_t *tpm_ek_tbs, size_t *tpm_ek_tbs_size) {
+  hmac_digest_t digest;
+
+  tpm_ek_tbs_values_t tpm_ek_tbs_params = {
+      .auth_key_key_id = inputs->auth_key_key_id,
+      .auth_key_key_id_size = kDiceCertKeyIdSizeInBytes,
+      .creator_pub_key_ec_x = (uint8_t *)tpm_ek_pubkey->x,
+      .creator_pub_key_ec_x_size = kAttestationPublicKeyCoordBytes,
+      .creator_pub_key_ec_y = (uint8_t *)tpm_ek_pubkey->y,
+      .creator_pub_key_ec_y_size = kAttestationPublicKeyCoordBytes,
+      .creator_pub_key_id = (uint8_t *)digest.digest, // Nobody cares.
+      .creator_pub_key_id_size = sizeof(digest),
+      .sal_version = "0.0.1",
+      .sal_version_len = 5,
+      .sal_manufacturer = "Nuvoton",
+      .sal_manufacturer_len = 7,
+      .sal_model = "Ti50",
+      .sal_model_len = 4,
+  };
+
+  HARDENED_RETURN_IF_ERROR(
+      tpm_ek_build_tbs(&tpm_ek_tbs_params, tpm_ek_tbs, tpm_ek_tbs_size));
+
```

Comment:
What do we need to do with the private key? should these be stored somewhere so they are accessible by owner firmware later? Should we store them in OTBN DMEM along side the DICE CDI_1 private key?

Reply:
I was under impression that the private key is not stored anywhere and is generated at run time when necessary?

But the whole issue of storing the generated certificates is open, let's discuss it once we get the generation going.